### PR TITLE
[SYCL] Don't destroy fence if kernel submission failed

### PR DESCRIPTION
**Pull request by againull**
This is a workaround for the bug in the L0 runtime. Currently if a command list submission failed then destruction of the associated fence with result in a hang. For now avoid destructing a fence if cmd list submission failed.

### CODE REVIEW

File: sycl/plugins/level_zero/pi_level_zero.hpp [hunk 1]
```diff
@@ -419,6 +419,14 @@ struct pi_command_list_info_t {
   // TODO: use this for optimizing events in the same command-list, e.g.
   // only have last one visible to the host.
   std::vector<pi_event> EventList{};
+
+  // TODO: remove when the gpu runtime is fixed.
+  // If an error happened at the point of a kernel submission then the
+  // associated fence is not in a valid state. Currently a program hangs if
+  // zeFenceDestroy is called for such fence. This bool variable allows to avoid
+  // the fence destruction if a kernel submission failed.
+  bool IsValidFence{true};
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.hpp:428]: I have just today submitted a patch for a similar reason, avoid sync of queue that had a failed submission on it: https://github.com/intel/llvm/pull/5524/files#diff-1f19baf7cc76d94bf4a1819c81e7b34292fb3ce215cd50873e8c9236cc73601cR808

Are you sure that it is zeFenceDestroy that is problematic in your case, not the sync of the queue?
And even then, can we maybe avoid destroy of all fences on the non-healthy queue since this is an error condition anyways?

Comment (againull) [reply sycl/plugins/level_zero/pi_level_zero.hpp:428]: oh, I see, sorry, I've just saw your PR. Yes, I am sure that zeFenceDestroy causes a hang in my case (queue sync causes hang as well, but avoiding queue sync is not enough).
I will rework this PR on top of yours.

>And even then, can we maybe avoid destroy of all fences on the non-healthy queue since this is an error condition anyways?
Ok, I will do so.

**Comment by againull** [comment]
Pls review only the second commit.

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 2]
```diff
@@ -865,10 +865,13 @@ pi_result _pi_queue::resetCommandList(pi_command_list_ptr_t CommandList,
   // synchronized above already does that.
   auto &EventList = CommandList->second.EventList;
   for (auto &Event : EventList) {
+    // All events in this loop are in the same command list which has been just
+    // reset above. We don't want cleanup() to reset same command list again for
+    // all events in the loop so set it to nullptr.
+    Event->ZeCommandList = nullptr;
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.cpp:871]: could you move this into a separate PR please?

Comment (againull) [reply sycl/plugins/level_zero/pi_level_zero.cpp]: Yes, sure.

**Review by smaslov-intel — APPROVED** [review]
